entity_extractor.py

```diff
@@ -0,0 +1,72 @@
+"""NER Entity Token Classifier"""
+from typing import Dict, List, Any
+from pydantic import Field
+from transformers import (
+  pipeline,
+  AutoTokenizer,
+  AutoModelForTokenClassification)
+
+class EntityExtractor():
+  """Entity Extractor"""
+
+  _input_text: str = Field(default=None)
+  _predictions: List[Dict[str, Any]] = Field(default=[])
+  _locations: List[str] = Field(default=[])
+  _organsiations: List[str] = Field(default=[])
+  _persons: List[str] = Field(default=[])
+
+  def __init__(
+    self,
+    model_name:str="Jean-Baptiste/roberta-large-ner-english"):
+
+
+    self._model_name = model_name
+
+    tokenizer = AutoTokenizer.from_pretrained(model_name)
+
+    model = AutoModelForTokenClassification.from_pretrained(model_name)
+
+    self._model = pipeline('ner', model=model, tokenizer=tokenizer, aggregation_strategy="simple")
+
+  def inference(self, input_text:str) -> Dict:
+    """inference function"""
+
+    self._input_text = input_text
+
+    self._predictions = self._model(self._input_text)
+
+    return self._predictions
```

Comment:
<img src="https://avatars.githubusercontent.com/u/116506391?s=400&u=09ae00df628c1f06039a246d04cccb2b4042ed0b&v=4" alt="Image description" width="20" height="20">   CodeSweep

The `inference` method modifies the state of the object by changing the `_input_text` and `_predictions` attributes. This could lead to unexpected behavior if multiple inferences are made with the same object. Consider making this method stateless by returning the predictions directly and not storing them in the object. If you need to store the results for later use, consider creating a separate method for that purpose.

```diff
-  def inference(self, input_text:str) -> Dict:
-    """inference function"""
-    self._input_text = input_text
-    self._predictions = self._model(self._input_text)
-    return self._predictions

+  def inference(self, input_text:str) -> Dict:
+    """inference function"""
+    predictions = self._model(input_text)
+    return predictions
```


<!-- This is an auto-generated comment by OSS CodeSweep -->